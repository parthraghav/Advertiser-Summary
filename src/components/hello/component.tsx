import React, { FunctionComponent } from "react";
import "./styles.scss";

// // // //

export const Hello: FunctionComponent = () => {
    return (
        <div className="row">
            <div className="col-lg-12 text-center">
                <p className="lead mb-0">Example Extension</p>
            </div>
        </div>
    );
};
