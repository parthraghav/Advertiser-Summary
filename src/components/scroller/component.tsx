import React, { FunctionComponent } from "react";
import "./styles.scss";
import { browser, Tabs } from "webextension-polyfill-ts";

// // // //

// Scripts to execute in current tab
const scrollToTopScript = `window.scroll(0,0)`;
const scrollToBottomScript = `window.scroll(0,9999999)`;

/**
 * Executes a string of Javascript on the current tab
 * @param code The string of code to execute on the current tab
 */
function executeScript(code: string): void {
    // Query for the active tab in the current window
    browser.tabs
        .query({ active: true, currentWindow: true })
        .then((tabs: Tabs.Tab[]) => {
            // Pulls current tab from browser.tabs.query response
            const currentTab: Tabs.Tab | undefined = tabs[0];

            // Short circuits function execution is current tab isn't found
            if (!currentTab) {
                return;
            }

            // Executes the script in the current tab
            browser.tabs
                .executeScript(currentTab.id, {
                    code,
                })
                .then(() => {
                    console.log("Done Scrolling");
                });
        });
}

// // // //

export const Scroller: FunctionComponent = () => {
    return (
        <div className="row">
            <div className="col-lg-12">
                <button
                    className="btn btn-block btn-outline-dark"
                    onClick={(): void => executeScript(scrollToTopScript)}
                >
                    Scroll To Top
                </button>
                <button
                    className="btn btn-block btn-outline-dark"
                    onClick={(): void => executeScript(scrollToBottomScript)}
                >
                    Scroll To Bottom
                </button>
            </div>
        </div>
    );
};
