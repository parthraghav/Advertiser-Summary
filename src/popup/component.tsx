import React, { FunctionComponent } from "react";
import { Hello } from "@src/components/hello";
import { browser } from "webextension-polyfill-ts";
import { Scroller } from "@src/components/scroller";
import "./styles.scss";

// // // //

export const Popup: FunctionComponent = () => {
    // Sends the `popupMounted` event
    React.useEffect(() => {
        browser.runtime.sendMessage({ popupMounted: true });
    }, []);

    // Renders the component tree
    return (
        <div className="popup-container">
            <div className="container mx-4 my-4">
                <Hello />
                <hr />
                <Scroller />
            </div>
        </div>
    );
};
