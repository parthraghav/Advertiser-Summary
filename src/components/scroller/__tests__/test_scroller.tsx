import * as React from "react";
import { Scroller } from "../component";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";

it("component renders", () => {
    const tree: ReactTestRendererJSON | null = renderer
        .create(<Scroller />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
