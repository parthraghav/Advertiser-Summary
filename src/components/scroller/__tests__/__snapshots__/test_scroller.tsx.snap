// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-12"
  >
    <button
      className="btn btn-block btn-outline-dark"
      onClick={[Function]}
    >
      Scroll To Top
    </button>
    <button
      className="btn btn-block btn-outline-dark"
      onClick={[Function]}
    >
      Scroll To Bottom
    </button>
  </div>
</div>
`;
